### DISCUSSION TITLE
Geolocation Javascript

### DISCUSSION THREAD

**Discussion by mister-vini** [frontendbr/forum#1058] [Perguntas e Respostas]
Boa noite pessoal, saberiam me dizer como poderia resolver um problema de geolocation, eu estou fazendo no navagation do javascript, estou tendo um problema de recebimento de position, tem hora que eu me retorna, mas a maioria das vezes não me retorna, ja tentei trocar o navagador, mas também não é, eu conversei com uma pessoa e ela falou que poderia ser o protocolo HTTP, seria HTTPS, como faria isso se eu estiver no ip local? Obrigado pela atenção, boa noite novamente!

**Comment by Morette** [comment]
se estiver usando a API do Google Maps, aqui tem um tuto legal que dá pra seguir

https://developers.google.com/maps/documentation/javascript/adding-a-google-map

**Comment by thulioph** [comment]
O [`navigator.geolocation`](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/Using_geolocation) considera o **localhost** como um ["contexto seguro"](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts#When_is_a_context_considered_secure). 
Em teoria era pra funcionar, estranho você não tá conseguindo. 
Tenta executar o código num serviço, tipo o [Codepen](https://codepen.io/), quem sabe ajuda a descobrir onde está o erro.
